modules/core/elb.tf

```diff
@@ -177,7 +177,7 @@ resource "aws_security_group_rule" "nomad_http" {
 # A Record for nomad API endpoint to point to Internal Load balancer
 resource "aws_route53_record" "nomad_rpc" {
   zone_id = data.aws_route53_zone.default.zone_id
-  name    = var.nomad_api_domain
+  name    = var.nomad_subdomain
```

Comment:
I see that the change is done in order to separate both the domain in Route53 record and domain for the actual Consul/Nomad/Vault service

Though since previously it was presumed that the domain in Route53 directly cooresponds to the actual service domain, and this repo is a public-facing one (i.e. prefer to prevent breaking changes), can I suggest the following:

Make `var.nomad_subdomain` optional, and if provided, it takes precedence over `var.nomad_api_domain`:

```tf
name = var.nomad_subdomain != "" ? var.nomad_subdomain : var.nomad_api_domain
```

Same for the other `xxx_subdomain` variables.

It might help to rename `nomad_subdomain` to `nomad_route53_subdomain` or something to more strongly indicate "the var acts as an override if provided" perhaps?

Reply:
Good point, edited to allow the use of the original var / avoid breaking changes